lib/Sema/CodeSynthesis.cpp

```diff
@@ -367,10 +367,6 @@ createCoroutineAccessorPrototype(AbstractStorageDecl *storage,
   if (isStatic)
     accessor->setStatic();
 
-  // The accessor is final if the storage is.
-  if (storage->isFinal())
-    makeFinal(ctx, accessor);
```

Comment:
makeFinal() should be dead code now, you can remove it from its .cpp and .h files